adapter/internal/operator/synchronizer/synchronizer.go

```diff
@@ -159,23 +179,26 @@ func sendAPIToAPKMgtServer(apiEvent APIEvent) {
 	})
 	if err != nil {
 		loggers.LoggerAPKOperator.ErrorC(logging.ErrorDetails{
-			Message:   fmt.Sprintf("error sending API to APK management server:%v", err),
+			Message:   fmt.Sprintf("Error sending API to APK management server : %v", err),
 			ErrorCode: 6001,
 			Severity:  logging.MAJOR,
 		})
 	}
-	loggers.LoggerAPKOperator.Info(res)
 }
 
 // getResourcesForAPI returns []*apiProtos.Resource for HTTPRoute
 // resources. Temporary method added until a proper implementation is done.
 func getResourcesForAPI(api APIState) []*apiProtos.Resource {
 	var resources []*apiProtos.Resource
 	var hostNames []string
-	for _, hostName := range api.ProdHTTPRoute.Spec.Hostnames {
+	httpRoute := api.ProdHTTPRoute
+	if httpRoute == nil {
+		httpRoute = api.SandHTTPRoute
```

Comment:
can both sandbox and prod route are nil? if not there can be nil pointer error at line no 201

Reply:
no, we do validate it in a previous step. https://github.com/wso2/apk/pull/277/files#diff-4a007a22d42a2f6c2f66ad83d13f3ba1c782bff21b193237800992f8f931f32cL93